### ISSUE TITLE
Efficient EHZ calculation

### ISSUE THREAD

**Issue by JoernStoehler** [JoernStoehler/msc-math-viterbo#6]
Motivation
- We want to sweep through a large number of polytopes and calculate their capacity, to see if we can find (new) counterexamples to Viterbo's conjecture.
- For this we need an efficient calculation method
- It's worth thinking briefly about batching (either with static num_facet or with facet padding; dimension can be assumed constant)
- We can validate our optimized implementation against our current unoptimized but trusted version
- Profiling ofc is needed to not optimize prematurely
- If truly necessary, you can add jax for further optimization via tracing (jax.jit); this will make some control flow more difficult though, so evaluate feasability of batching/using vmap

Proposal
- start with setting up profiling, profile our current implementation
- consider how to improve performance for a single call, and for batched calls (i.e. whether vmap is possible via padding / via demanding matching num_facet counts etc)
- then add a new file with an optimized algorithm for a single call
- add unit tests to compare the new algo against the old algo wrt accuracy
- profile, tweak and optimize the new algo until you are satisfied
- Optional: you can add and use jax if that is something you believe is feasible and useful; we will add jax eventually for other parts of the MSc thesis anyway, so this doesn't add dev complexity 

If you have own ideas, please go ahead! I want to see what you do with the original motivation, my proposals are just suggestions.
